x-pack/plugins/apm/public/components/alerting/ui_components/chart_preview/index.tsx

```diff
@@ -64,36 +67,89 @@ export function ChartPreview({
     opacity: thresholdOpacity,
   };
 
+  const NUM_SERIES = 5;
+  const DEFAULT_DATE_FORMAT = 'Y-MM-DD HH:mm:ss';
+
+  const tooltipProps: TooltipProps = {
+    headerFormatter: ({ value }) => {
+      const dateFormat =
+        (uiSettings && uiSettings.get(UI_SETTINGS.DATE_FORMAT)) ||
+        DEFAULT_DATE_FORMAT;
+      return moment(value).format(dateFormat);
+    },
+  };
+
+  const filteredSeries = useMemo(() => {
+    const sortedSeries = series.sort((a, b) => {
+      const aMax = Math.max(...a.data.map((point) => point.y as number));
+      const bMax = Math.max(...b.data.map((point) => point.y as number));
+      return bMax - aMax;
+    });
+    return sortedSeries.slice(0, NUM_SERIES);
+  }, [series]);
+
+  const barSeries = useMemo(() => {
+    return filteredSeries.reduce<
+      Array<{ x: number; y: Maybe<number>; groupBy: string | undefined }>
+    >((acc, serie) => {
+      const barPoints = serie.data.reduce<
+        Array<{ x: number; y: Maybe<number>; groupBy: string | undefined }>
+      >((pointAcc, point) => {
+        return [...pointAcc, { ...point, groupBy: serie.name }];
+      }, []);
+      return [...acc, ...barPoints];
+    }, []);
+  }, [filteredSeries]);
+
+  const timeZone = getTimeZone(uiSettings);
+
+  const legendSize =
+    filteredSeries.length > 1
+      ? Math.ceil(filteredSeries.length / 2) * 30
+      : filteredSeries.length * 35;
+
+  const chartSize = 150;
+
+  const { yMin, yMax, xMin, xMax } = getDomain(filteredSeries);
+  const chartDomain = {
+    max: Math.max(yMax, threshold) * 1.1, // Add 10% headroom.
+    min: Math.min(yMin, threshold),
+  };
+
+  if (chartDomain.min === threshold) {
+    chartDomain.min = chartDomain.min * 0.9; // Allow some padding so the threshold annotation has better visibility
+  }
+
+  const dateFormatter = useDateFormatter(xMin, xMax);
+
+  const lookback = timeSize * BUCKET_SIZE;
```

Comment:
You can calculate this from the `series`:

```suggestion
  const lookback = moment.duration(xMax - xMin).asMinutes();
```

Reply:
The lookback is not always in minutes. I think `const lookback = timeSize * BUCKET_SIZE;` is much simpler and easy to understand.

Reply:
I'd like to leave `BUCKET_SIZE` as an implementation detail and not expose it. But in the interest of time, I think it's fine. Let's at least rename it so it's clear that it's not a global bucket size for all alerting needs, but specifically for the previews:

```suggestion
  const lookback = timeSize * ALERT_PREVIEW_BUCKET_SIZE;
```